tasks/need_fixing_linters/golangci_lint_parser.py

```diff
@@ -0,0 +1,125 @@
+"""
+This module defines a parser for golangci-lint output.
+"""
+
+import re
+import os
+from codeowners import CodeOwners
+
+# example lint message
+# "pointer.go:6:1: package-comments: should have a package comment (revive)"
+LINT_PATTERN = re.compile("^([^:]+):([0-9]+):([0-9]+): (([^:]+): )?(.+) \\((.+)\\)$")
+
+# Example module message
+# "Linters for module /Users/pierre.gimalac/go/src/github.com/DataDog/datadog-agent/pkg/remoteconfig/state failed (base flavor)"
+MODULE_PATTERN = re.compile("^.*Linters for module ([^ ]+) failed \\((.+)\\).*$")
+
+CODEOWNERS_FILE_PATH = ".github/CODEOWNERS"
+
+def read_owners(owners_file: str):
+    """
+    Parses the CODEOWNERS file.
+    """
+    with open(owners_file, 'r', encoding="utf-8") as f:
+        codeowners = CodeOwners(f.read())
+    return codeowners
+
+
+def parse_file(golangci_lint_output: str):
+    """
+    Parses the output of the golangci-lint run.
+    Returns a Dict(linter: List(base_path, row, col, lint, descr)).
+    """
+    lints = []
+    current_module = None
+    for line in golangci_lint_output.split("\n"):
+        line = line.strip()
+
+        match = re.match(MODULE_PATTERN, line)
+        if match:
+            module_path, flavor = match.groups()
+            current_module = module_path
+            continue
+
+        if current_module is None:
+            continue
+
+        match = re.match(LINT_PATTERN, line)
+        if match is None:
+            continue
+
+        file_path, row, col, lint, _, descr, linter = match.groups()
+        full_path = os.path.normpath(os.path.join(current_module, file_path))
+        base_path = full_path.removeprefix(os.getcwd()).removeprefix('/')
+        lints.append((base_path, row, col, lint, descr, linter))
+
+    lints_per_linter = dict()
+    for base_path, row, col, lint, descr, linter in lints:
+        if linter not in lints_per_linter:
+            lints_per_linter[linter] = []
+        lints_per_linter[linter].append((base_path, row, col, lint, descr))
+
+    ## Uncomment to get the number of lints per linter
+    # for linter, lints in lints_per_linter.items():
+    #     print(f"{linter}: {len(lints)}")
+
+    return lints_per_linter
+
+
+def filter_lints(lints_per_linter, filter_team: str=None, filter_linters: str=None):
+    """
+    Keeps only the lints from a specific team and specific linters.
+
+        Parameters:
+            filter_team (str): Keep only the lints from the files owned by the filter_team. None will keep lints from all teams.
+            filter_linters (str): Comma-separated linters to keep. None will keep lints from all linters.
+    """
+    list_filter_linters = filter_linters.split(',') if filter_linters else []
+    filtered_lints = dict()
```

Comment:
same here, `defaultdict` should simplify your code

Reply:
Again, great feature ! Fixed in [8a4ad8d](https://github.com/DataDog/datadog-agent/pull/19015/commits/8a4ad8df6bc15855388760fd383eb8ca25948ab0).